python/graphstorm/trainer/np_trainer.py

```diff
@@ -217,7 +217,11 @@ def fit(self, train_loader, num_epochs,
         if self.rank == 0 and self.evaluator is not None:
             output = {'best_test_score': self.evaluator.best_test_score,
                        'best_val_score': self.evaluator.best_val_score,
-                       'peak_mem_alloc_MB': th.cuda.max_memory_allocated(device) / 1024 / 1024}
+                       'peak_mem_alloc_MB': th.cuda.max_memory_allocated(device) / 1024 / 1024,
+                       'best validation iteration': \
+                           self.evaluator.best_iter_num[self.evaluator.metric[0]],
```

Comment:
You need to check whether no-validation is set.

Reply:
ditto